tests/dummy/app/pods/demo/controller.js

```diff
@@ -1,51 +1,29 @@
 import Ember from 'ember'
+import _ from 'lodash'
 
 export default Ember.Controller.extend({
-  data: [
-    {
-      'value': 'Clark Kent',
-      'label': 'Superman'
-    },
-    {
-      'value': 'Bruce Wayne',
-      'label': 'Batman'
-    },
-    {
-      'value': 'Tony Starks',
-      'label': 'Ironman'
-    },
-    {
-      'value': 'Princess Diana',
-      'label': 'Wonder Woman'
-    },
-    {
-      'value': 'Barry Allen',
-      'label': 'The Flash'
-    },
-    {
-      'value': 'Arthur Curry',
-      'label': 'Aquaman'
-    },
-    {
-      'value': 'Hal Jordan',
-      'label': 'Green Lantern'
-    },
-    {
-      'value': 'Adam Meadows',
-      'label': 'Atom'
-    },
-    {
-      'value': 'Jaime Reyes',
-      'label': 'Blue Beetle'
-    },
-    {
-      'value': 'Leonard Snart',
-      'label': 'Captain Cold'
+  data: Ember.computed('data', 'search', function () {
```

Comment:
May I recommend using `ember-computed-decorators` which changes this to look something like:
```js
import computed, {readOnly} from 'ember-computed-decorators'

…

@readOnly
@computed('data', 'search')
data (data, search) {
  let result = this.model.map(record => {
    return {
      label: record.get('label'),
      value: record.id
    }
  })

  if (search) {
    return _.filter(result, item => {
      return item.label.toLowerCase().indexOf(this.get('search').toLowerCase()) !== -1
    })
  }

  return result
},
```